scheduler/leader.go

```diff
@@ -70,12 +70,18 @@ func Lead(shutdown <-chan interface{}, conn coordclient.Connection, cpClient dao
 // service has an address assignment the host will already be selected. If not
 // the host with the least amount of memory committed to running containers will
 // be chosen.  Returns the hostid, hostip (if it has an address assignment).
-func (l *leader) SelectHost(s *service.Service) (string, error) {
+func (l *leader) SelectHost(sn *zkservice.ServiceNode) (string, error) {
```

Comment:
I've added the appropriate fields (CPUCommitment, HostPolicy, and AddressAssignment) to ServiceNode. 

After looking at the code some more, I modified the StrategyService struct to have a zkservice.ServiceNode, rather than a Service object. Naming the fields properly made its implementation of the strategy.ServiceConfig interface work. Since it was only being used in StrategySelectHost, it's a safe change.

I also made the change you suggested of moving the code that gets the AddressAssignment from Service.Endpoints into NewServiceNodeFromService() (in zzk/service/service.go)